build/configuration-properties-collector/src/main/java/org/hibernate/search/configuration/properties/collector/ConfigurationPropertyCollector.java

```diff
@@ -0,0 +1,197 @@
+		if ( annotation.flatMap( a -> a.attribute( "ignore", boolean.class ) ).orElse( false ) ) {
+			return;
+		}
+
+		// Try to find a default value. Assumption is that the settings class has an inner class called "Defaults" and
+		// the key for the default value is exactly the same as the config constant name:
+		Object value = findDefault( constant );
+
+		properties.put(
+				constant.getEnclosingElement().toString() + "#" + constant.getSimpleName().toString(),
+				new ConfigurationProperty()
+						.javadoc( extractJavadoc( constant ) )
+						.key(
+								extractKey(
+										constant,
+										classPrefix,
+										annotation.flatMap( a -> a.attribute( "prefix", String.class ) )
+												.map( ConfigurationPropertyCollector::nullIfBlank )
+								)
+						)
+						.sourceClass( constant.getEnclosingElement().toString() )
+						.type( extractType( constant, annotation ) )
+						.defaultValue( value )
+		);
+	}
+
+
+	private String extractKey(VariableElement constant, Optional<String> classPrefix, Optional<String> constantPrefix) {
+		String prefix;
+		if ( constantPrefix.isPresent() ) {
+			prefix = constantPrefix.get();
+		}
+		else if ( classPrefix.isPresent() ) {
+			prefix = classPrefix.get();
+		}
+		else {
+			prefix = configPrefix;
+		}
+
+		return prefix + Objects.toString( constant.getConstantValue(), "NOT_FOUND#" + constant.getSimpleName() );
+	}
+
+	private HibernateSearchConfiguration.Type extractType(VariableElement constant, Optional<AnnotationUtils.AnnotationAttributeHolder> annotation) {
+		HibernateSearchConfiguration.Type type = annotation.flatMap( a -> a.attribute( "type", HibernateSearchConfiguration.Type.class ) )
+				.orElse( HibernateSearchConfiguration.Type.AUTO );
+		if ( HibernateSearchConfiguration.Type.AUTO.equals( type ) ) {
+			return constant.getEnclosingElement().getEnclosingElement()
+					.toString().contains( ".spi" ) ?
+					HibernateSearchConfiguration.Type.SPI :
+					HibernateSearchConfiguration.Type.API;
+		}
+		return type;
+	}
+
+	private String extractJavadoc(VariableElement constant) {
+		try {
+			Element enclosingClass = constant.getEnclosingElement();
+			Path docs = javadocsLocation.resolve(
+					enclosingClass.toString().replace( ".", File.separator ) + ".html"
+			);
+
+			// calling getEnclosingElement() on class should return a package:
+			String packagePath = enclosingClass.getEnclosingElement().toString().replace( ".", File.separator );
+
+			Document javadoc = Jsoup.parse( docs.toFile() );
+
+			org.jsoup.select.Elements blocks = javadoc.select( "#" + constant.getSimpleName() + " .block" );
+			if ( !blocks.isEmpty() ) {
+				org.jsoup.nodes.Element block = blocks.get( 0 );
+				for ( org.jsoup.nodes.Element link : block.getElementsByTag( "a" ) ) {
+					// only update links if they are not external:
+					if ( !link.hasClass( "external-link" ) ) {
+						link.attr( "href", javadocsBaseLink + packagePath + "/" + link.attr( "href" ) );
+					}
+				}
```

Comment:
Not sure how this works, but some links seem to be incorrect. See the documentation for `hibernate.search.coordination.entity.mapping.outboxevent.catalog`, for example.